### ISSUE TITLE
IndexError

### ISSUE THREAD

**Issue by luoxxib** [Vezarachan/GeoEvolve#2]
I conducted an experiment using my own data, with 125 samples in the training set, 16 in the validation set, and 16 in the test set. An error occurred while running initial_program.py. During execution, line 93 of the file contains the calculation:
q_level = np.ceil((1 - self.miscoverage_level) * (N + 1)) / N
(For my data, the calculated N is 16). This resulted in a q_level value of 1. Consequently, in line 57:
idx = np.sum(normalized_cumulative_weights <= q, axis=1)
the calculated idx values all became 16. However, scores_sorted is a 16x16 matrix. This caused an out-of-bounds error in line 58:
quantiles = scores_sorted[np.arange(N_test), idx]
since the indices exceed the valid range for the matrix dimensions.
